source/documentation/tools/create-a-derived-table/project-structure.md

```diff
@@ -1,66 +1,723 @@
+
+### Staging: Files and folders
+
+Let's zoom into the staging directory from our `models` file tree in the overview section and walk through what's going on here.
+
+```shell
+models/staging
+├── stg_nomis
+│   ├── stg_nomis__docs.md
+│   ├── stg_nomis__models.yml
+│   ├── stg_nomis__offender.sql
+│   └── stg_nomis__prison.sql
+└── stg_xhibit
+    ├── stg_xhibit__models.yml
+    └── stg_xhibit__court.sql
+```
+
+- **Folders.** Folder structure is extremely important in dbt. Not only do we need a consistent structure to find our way around the codebase, as with any software project, but our folder structure is also one of the key interfaces for understanding the knowledge graph encoded in our project (alongside the [DAG](https://en.wikipedia.org/wiki/Directed_acyclic_graph) and the data output into our warehouse). It should reflect how the data flows, step-by-step, from a wide variety of source-conformed models into fewer, richer business-conformed models. Moreover, we can use our folder structure as a means of selection in dbt [selector syntax](https://docs.getdbt.com/reference/node-selection/syntax). For example, with the above structure, if we got fresh xhibit data loaded and wanted to run all the models that build on our xhibit data, we can easily run `dbt build --select models/staging/xhibit_stg+`, and we’re all set for building more up-to-date reports on Crown Courts. Additionally, with create-a-derived-table the project structure is how we manage database access. So following this guidance will ensure only those who need to get to see your data get to see it.
+  - ✅ **Subdirectories based on business grouping.** dbt recommends against this practice, however crate-a-derived-table has been built in a way that necessitates domains as subdirectories so that we can control access through [data engineering database access](https://github.com/moj-analytical-services/data-engineering-database-access/tree/main/database_access/create_a_derived_table). This is a key deviation from dbt guidance.
+  - ✅ **Subdirectories based on the source system**. For the MoJ we have a staging domain and each subfolder in the domain will form its own database. We expect that each database in the staging domain should be source aligned. This allows for future projects to pull on data that is already staged, saving time and reducing the chance of silo-ing data.
+  - ❌ **Subdirectories based on loader.** Some people attempt to group by how the data is loaded (Fivetran, Stitch, custom syncs), but this is too broad to be useful on a project of any real size.
```

Comment:
This might need some thought / tweaking to suit our specific setup - I'm thinking specifically:
- People might not be familiar with Fivetran or Stitch to understand this
- More relevant to us would be dlt vs create-a-pipeline vs data uploader